### PR TITLE
rm: refactor prompt_file()

### PR DESCRIPTION
**Pull request by cakebaker**
This PR applies two refactorings to `prompt_file()` to simplify the function:

- extract a `prompt_dir()` function
- replace an `if` with a `match`

### CODE REVIEW

File: src/uu/rm/src/rm.rs [hunk 1]
```diff
@@ -437,8 +437,23 @@ fn remove_file(path: &Path, options: &Options) -> bool {
...
+    // We can't use metadata.permissions.readonly for directories because it only works on files
+    // So we have to handle whether a directory is writable manually
+    if let Ok(metadata) = fs::metadata(path) {
+        handle_writable_directory(path, options, &metadata)
+    } else {
+        true
+    }
+}
+
 #[allow(clippy::cognitive_complexity)]
```
Comment (tertsdiepraam) [inline src/uu/rm/src/rm.rs:455]: Maybe we can remove this now?

**Review by tertsdiepraam — APPROVED** [review]
One suggestion, looks good otherwise!

File: src/uu/rm/src/rm.rs (hunk 1 above)
Comment (cakebaker) [reply src/uu/rm/src/rm.rs:455]: No, there is still a warning from the `clippy::cognitive_complexity` lint :|

Comment (tertsdiepraam) [reply src/uu/rm/src/rm.rs:455]: :(

**Pull request merged**